velox/exec/prefixsort/PrefixSortEncoder.h

```diff
@@ -51,6 +50,48 @@ class PrefixSortEncoder {
     }
   }
 
+  /// Encodes String types.
+  /// The first byte of the encoded result is null byte. The value is 0 if
+  /// (nulls first and value is null) or (nulls last and value is not null).
+  /// Otherwise, the value is 1.
+  FOLLY_ALWAYS_INLINE void encode(
+      std::optional<StringView> value,
+      char* dest,
+      int32_t encodeSize) const {
+    if (value.has_value()) {
+      dest[0] = nullsFirst_ ? 1 : 0;
+      encodeNoNulls(value.value(), dest + 1, encodeSize - 1);
+    } else {
+      dest[0] = nullsFirst_ ? 0 : 1;
+      std::memset(dest + 1, 0, encodeSize - 1);
+    }
+  }
+
+  /// The string prefix is formatted as 'null byte + string content + padding
+  /// zeros'. If `!ascending_`, the bits for both the content and padding zeros
+  /// need to be inverted.
+  FOLLY_ALWAYS_INLINE void encodeNoNulls(
```

Comment:
This function could be moved to PrefixSortEncoder.cpp though other function such as `detail::encodeDouble` should also move but not create this new file to move to.

Reply:
I think by now we may not need to extract encodeNoNulls function for string type.
It's better to move them to cpp file, but maybe do it in a separate PR.